Update headers and credits

Zimmi48 opened the pull request:
This pull request contains several proposals related to the credits:
- I removed the redundant `COPYRIGHT` file (its content is just a duplicate of `CREDITS`) and explicitly referred to `CREDITS` from the copyright headers (for now, I'm just changing the header template but this would require a larger change to all source files which could be done at the same time as the update to the copyright years).
- I also removed the list of research labs in the headers: first of all, it doesn't make much sense to list them there (they cannot be copyright holders being neither the authors, nor their employers), the list is necessarily incomplete, already obsolete (PPS was merged into IRIF in 2016). I added a general mention "and contributors" which is very classic; together with the reference to the `CREDITS` file, this makes the headers more accurate. I kept INRIA and CNRS but maybe it should be further reduced to INRIA and contributors (INRIA being the employer of the most part of the development team, with other prominent members of the development team having been attached to INRIA as members of one of its project-teams).
- I've added a paragraph in the contributing guide suggesting that people add themselves to the `CREDITS` file (this file was very badly maintained in the most recent years, so this should help avoiding that). I'm also suggesting people to add their institution (as was already the case for the people already listed). This will allow to add this information to the Zenodo page: https://zenodo.org/record/1133970#.Wky2xBYo9hF (for now I've only added the institutions I knew about or could find easily). I'm also suggesting to people to add their ORCID (if they have one). If people provide this information, it can be added to the Zenodo page (and this allows to have a link from the contributor's name to their personal page).

Finally, this PR also contains a new update to `CREDITS`.

maximedenes commented:
@herbelin does this look good to you?

@gares what about the SSR part?

gares commented:
I think it should contain the same data that was there before, and it is not the case.

First the plugins are two, second the authors of the two plugins are exactly the 3 listed (no `...`), finally the institute was the MSR-Inria JC.

gares commented:
Indeed @Zimmi48  has a `[ ]` item for ssr, so I guess I'll add a label

Zimmi48 commented:
@gares 
> I think it should contain the same data that was there before, and it is not the case.

Sorry I didn't understand if your comment applies to the whole PR or just answers Maxime's question. In particular, SSR was not mentioned before. I added it.

> First the plugins are two

You mean ssreflect + ssrmatching? I did not remove ssrmatching.

> second the authors of the two plugins are exactly the 3 listed (no ...)

OK. I added these names and the ... because I was not sure. But you're saying that Assia was missed in the credits for ssrmatching as well, right?

> finally the institute was the MSR-Inria JC

Can you be a bit more specific: is it the institution of the three authors? What dates should I put for Assia in ssrmatching. What institution should I put for you in ssreflect. Right now, for ssrmatching it is:

```
   developed by Georges Gonthier (Microsoft Research - Inria Joint Centre, 2007-2011),
   and Enrico Tassi (Inria-Marelle, 2011-now)
```

Is this accurate?

Note: you can either give me enough info so that I can adapt my PR or you can even push a commit directly if you prefer.

Zimmi48 commented:
@gares Ping.

@herbelin Ping.

herbelin commented:
A few comments:
- Suggesting an ORCID, why not. Do our labs provide an ORCID for us?
- @zimmi48: your own credits line: I guess you should say PPS for 2015 vs IRIF: I believe we should keep the name PPS for the lab before 1/1/2017 (but we can add for clarity a remark that PPS was merged to LIAFA to become IRIF at the beginning of 2017)
- CoqHoTT: as an INRIA team, it was named Ascola then Galinette. I guess that whether PMP should be identified as employed by the ERC CoqHoTT project or as employed by INRIA is a subjective matter (@mattam82?)
- meaning of "significant": this shall be difficult to evaluate. I don't know what to suggest but significant could be measured in term of time spent, of impact for other users, ...
- removal of file `COPYRIGHT`: this seems reasonable since it is outdated or redundant information; I don't know if there are legal implications about this removal but I guess that if yes, we can justify that the word "copyright" is also used in headers, so this looks reasonable to me
- "for the complete list of authors": I would preventively and reasonably remove the word "complete"
- I'm actually afraid that we sometimes forget to acknowledge credits, we should be as fair as possible on this point
- I'm not a jurist and I'm not claiming that the above is meaninful from a legal point of view; I only tried to give personal comments which looks meaningful to me; I'm referring otherwise to @mattam82

Zimmi48 commented:
- Yes, you may log into ORCID by using an account from Inria or Université Paris-Diderot (many other institutions, like CNRS, are also represented).
- Significant: this is supposed to correspond to "copyrightable". Because this notion is not clearly defined, it is advised to be cautious and consider a large panel of contributions as significant (but fixing typos definitely does not count).
- There is no legal implication to not having a file named `COPYRIGHT` and most projects don't have one.
- OK for your other remarks.

Zimmi48 commented:
Pushed a revised version taking the various comments into account.
@gares We still need to do better for the reference of the authors of the ssrmatching and ssreflect plugin, in particular:
- Did @amahboubi contribute to ssrmatching?
- What were her institutions and at what dates did she contribute?
- What dates should we put for @ggonthier? The current dates implied he stopped working on ssreflect and ssrmatching in 2011? Is this accurate? It sounds like it should be a more recent date, shouldn't it?

psteckler commented on CREDITS:
```diff
@@ -159,7 +164,8 @@ of the Coq Proof assistant during the indicated time:
   Benjamin Werner (INRIA, 1989-1994)
   Nickolai Zeldovich (MIT 2014-2016)
-  Théo Zimmermann (INRIA-IRIF, 2015-now)
+  Théo Zimmermann (ORCID: https://orcid.org/0000-0002-3580-8806,
```
My name appears a few lines above. For that entry, you should change "now" to "2018". :-(

Zimmi48 replied:
OK :-(

Zimmi48 commented:
@gares Ping.

If I don't get a reply on the topic of SSReflect credits by Wednesday, I'll suggest merging without waiting further. Right now, SSReflect is completely missing in CREDITS so this PR can only improve on things.

Zimmi48 commented:
OK this has waited enough. I propose merging now. This will be followed by a PR updating the header in all the other files.

gares commented on CREDITS:
```diff
@@ -57,16 +62,14 @@ plugins/setoid_ring
   and Bruno Barras (INRIA LogiCal, 2005-2006),
+plugins/ssreflect
+  developed by Georges Gonthier (Microsoft Research - Inria Joint Centre, 2007-2011),
+  Assia Mahboubi and Enrico Tassi (Inria-Marelle, 2011-now).
```
I think putting the Inria team name is too unstable.  Assia was in the Inria joint center, typical, then specfun.  I was in the joint centre, then specfun and now marelle.

Just put Inria

Zimmi48 replied:
OK. What about the ending dates? Shouldn't it be at least 2017 for everyone (including George)?

Zimmi48 commented:
> Just put Inria

BTW I wondered about the usefulness of the current level of details for developers' institutions. It would be simpler to just put the employer, e.g. Inria, Université Paris-Diderot, CNRS, Université Paris-Sud, etc. (Even this is unstable, but possibly less unstable).

Zimmi48 commented:
FTR this is ready for merging. Further improvement to the Credits, like deciding to keep only the name of employer (see previous comment) can come in later PRs.
